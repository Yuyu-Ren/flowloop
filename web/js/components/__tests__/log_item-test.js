/* eslint-env jest */

import LogItem from '../log_item'

import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

describe('LogItem', () => {
  it('exists', () => {
    expect(LogItem).toBeTruthy()
  })

  it('renders', () => {
    const log = {
      id: '_log',
      timerType: 'work',
      startedAt: new Date('04/20/2010'),
      endedAt: new Date('04/20/2010'),
      duration: 60000,
      labelId: '_work',
      isComplete: true
    }

    const labels = {
      '_work': {
        id: '_work',
        name: 'Work',
        color: 'blue'
      }
    }

    const comp = <LogItem item={log} labels={labels} />
    const wrapper = shallow(comp)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
