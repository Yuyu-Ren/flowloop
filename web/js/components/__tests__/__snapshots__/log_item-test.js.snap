// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogItem renders 1`] = `
<span
  aria-label="Work - 12:00 am"
  className="timeline-small-item hint--top hint--no-animate -work"
>
  <span
    className="peg"
    style={
      Object {
        "backgroundColor": "blue",
      }
    }
  />
</span>
`;
